activesupport/lib/active_support/callbacks.rb

```diff
@@ -100,39 +100,40 @@ def run_callbacks(kind)
         env = Filters::Environment.new(self, false, nil)
         next_sequence = callbacks.compile
 
-        invoke_sequence = Proc.new do
-          skipped = nil
-          while true
-            current = next_sequence
-            current.invoke_before(env)
-            if current.final?
-              env.value = !env.halted && (!block_given? || yield)
-            elsif current.skip?(env)
-              (skipped ||= []) << current
-              next_sequence = next_sequence.nested
-              next
-            else
-              next_sequence = next_sequence.nested
-              begin
-                target, block, method, *arguments = current.expand_call_template(env, invoke_sequence)
-                target.send(method, *arguments, &block)
-              ensure
-                next_sequence = current
-              end
-            end
-            current.invoke_after(env)
-            skipped.pop.invoke_after(env) while skipped && skipped.first
-            break env.value
-          end
-        end
-
         # Common case: no 'around' callbacks defined
         if next_sequence.final?
           next_sequence.invoke_before(env)
           env.value = !env.halted && (!block_given? || yield)
           next_sequence.invoke_after(env)
           env.value
         else
+          invoke_sequence = Proc.new do
+            skipped = []
```

Comment:
What purpose is changed `skipped = nil` to `skipped = []`?

Reply:
The variable makes more sense being initialized as an array, in my opinion. This is how it is used in the code:

```ruby
skipped = nil
...
(skipped ||= []) << current
...
skipped.pop.invoke_after(env) while skipped && skipped.first
```

Which was changed to

```ruby
skipped =[]
...
skipped << current
...
skipped.pop.invoke_after(env) while skipped.any?
```

Reply:
I'd prefer less allocation as the original code.
And I don't want someone create a PR "#37969 lost the less allocation optimization, I want to recover the faster optimization".